src/textual/widgets/_option_list.py

```diff
@@ -0,0 +1,904 @@
+
+    BINDINGS: ClassVar[list[BindingType]] = [
+        Binding("down", "down", "Down", show=False),
+        Binding("end", "last", "Last", show=False),
+        Binding("enter", "select", "Select", show=False),
+        Binding("home", "first", "First", show=False),
+        Binding("page_down", "page_down", "Page Down", show=False),
+        Binding("page_up", "page_up", "Page Up", show=False),
+        Binding("up", "up", "Up", show=False),
+    ]
+    """
+    | Key(s) | Description |
+    | :- | :- |
+    | down | Move the highlight down. |
+    | end | Move the highlight to the last option. |
+    | enter | Select the current option. |
+    | home | Move the highlight to the first option. |
+    | page_down | Move the highlight down a page of options. |
+    | page_up | Move the highlight up a page of options. |
+    | up | Move the highlight up. |
+    """
+
+    COMPONENT_CLASSES: ClassVar[set[str]] = {
+        "option-list--option-disabled",
+        "option-list--option-highlighted",
+        "option-list--option-highlighted-disabled",
+        "option-list--option-hover",
+        "option-list--option-hover-disabled",
+        "option-list--option-hover-highlighted",
+        "option-list--option-hover-highlighted-disabled",
+        "option-list--separator",
+    }
+    """
+    | Class | Description |
+    | :- | :- |
+    | `option-list--option-disabled` | Target disabled options. |
+    | `option-list--option-highlighted` | Target the highlighted option. |
+    | `option-list--option-highlighted-disabled` | Target a disabled option that is also highlighted. |
+    | `option-list--option-hover` | Target an option that has the mouse over it. |
+    | `option-list--option-hover-disabled` | Target a disabled option that has the mouse over it. |
+    | `option-list--option-hover-highlighted` | Target a highlighted option that has the mouse over it. |
+    | `option-list--option-hover-highlighted-disabled` | Target a disabled highlighted option that has the mouse over it. |
+    | `option-list--separator` | Target the separators. |
+    """
+
+    DEFAULT_CSS = """
+    OptionList {
+        background: $panel-lighten-1;
+        color: $text;
+        overflow-x: hidden;
+    }
+
+    OptionList > .option-list--separator {
+        color: $text-muted;
+    }
+
+    OptionList > .option-list--option-highlighted {
+        background: $accent 50%;
+        color: $text;
+        text-style: bold;
+    }
+
+    OptionList:focus > .option-list--option-highlighted {
+        background: $accent;
+    }
+
+    OptionList > .option-list--option-disabled {
+        color: $text-disabled;
+    }
+
+    OptionList > .option-list--option-highlighted-disabled {
+        color: $text-disabled;
+        background: $accent 30%;
+    }
+
+    OptionList:focus > .option-list--option-highlighted-disabled {
+        background: $accent 40%;
+    }
+
+    OptionList > .option-list--option-hover {
+        background: $panel-lighten-2;
```

Comment:
Consider $boost for these states, which brightens or darkens depending on the start of dark mode.

Reply:
Nice idea, I always seem to forget about `$boost`. Done.